geonode/api/tests.py

```diff
@@ -226,3 +236,222 @@ def test_date_filter(self):
         resp = self.api_client.get(filter_url)
         self.assertValidJSONResponse(resp)
         self.assertEquals(len(self.deserialize(resp)['objects']), 4)
+
+
+class LayersStylesApiInteractionTests(LiveServerTestCase, ResourceTestCase):
+
+    """Test Layers"""
+
+    fixtures = ['initial_data.json', 'bobby']
+
+    def setUp(self):
+        super(LayersStylesApiInteractionTests, self).setUp()
+        # Reconnect signals for this Test Class
+
+        if check_ogc_backend(geoserver.BACKEND_PACKAGE):
+            reconnect_signals()
+
+        call_command('loaddata', 'people_data', verbosity=0)
+
+        self.layer_list_url = reverse(
+            'api_dispatch_list',
+            kwargs={
+                'api_name': 'api',
+                'resource_name': 'layers'})
+        self.style_list_url = reverse(
+            'api_dispatch_list',
+            kwargs={
+                'api_name': 'api',
+                'resource_name': 'styles'})
+        filename = os.path.join(gisdata.GOOD_DATA, 'raster/test_grid.tif')
+        self.layer = file_upload(filename)
+        all_public()
+
+    def tearDown(self):
+        Layer.objects.all().delete()
+
+        if check_ogc_backend(geoserver.BACKEND_PACKAGE):
+            disconnect_signals()
+
+    def test_layer_interaction(self):
+        """Layer API interaction check."""
+        layer_id = self.layer.id
+
+        layer_detail_url = reverse(
+            'api_dispatch_detail',
+            kwargs={
+                'api_name': 'api',
+                'resource_name': 'layers',
+                'id': layer_id
+            }
+        )
+        resp = self.api_client.get(layer_detail_url)
+        self.assertValidJSONResponse(resp)
+        obj = self.deserialize(resp)
+        # Should have links
+        self.assertTrue('links' in obj and obj['links'])
+        # Should have default style
+        self.assertTrue('default_style' in obj and obj['default_style'])
+        # Should have styles
+        self.assertTrue('styles' in obj and obj['styles'])
+
+        # Test filter layers by id
+        filter_url = self.layer_list_url + '?id=' + str(layer_id)
+        resp = self.api_client.get(filter_url)
+        self.assertValidJSONResponse(resp)
+        # This is a list url
+        objects = self.deserialize(resp)['objects']
+        self.assertEqual(len(objects), 1)
+        obj = objects[0]
+        # Should not have links (to save payload from big text)
+        self.assertTrue('links' not in obj)
+        # Should not have styles
+        self.assertTrue('styles' not in obj)
+        # Should have default_style
+        self.assertTrue('default_style' in obj and obj['default_style'])
```

Comment:
This means, if the request returns JSON object ```data``` you can get the default style from ```data['objects'][<index>]['default_style']```, where index is the index of layer in the objects list that you want.